runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -540,9 +541,25 @@ public void onChatMessage(ChatMessage event)
 			cannonTimer.setTooltip(cannonTimer.getTooltip() + " - World " + client.getWorld());
 		}
 
-		if (config.showCannon() && (message.equals(CANNON_PICKUP_MESSAGE) || message.equals(CANNON_DESTROYED_MESSAGE)))
+		if (config.showCannon())
 		{
-			removeGameTimer(CANNON);
+			if (message.equals(CANNON_FURNACE_MESSAGE) || message.contains(CANNON_REPAIR_MESSAGE))
+			{
+				removeGameTimer(CANNON_REPAIR);
+				TimerTimer cannonTimer = createGameTimer(CANNON);
+				cannonTimer.setTooltip(cannonTimer.getTooltip() + " - World " + client.getWorld());
+			}
+			else if (message.equals(CANNON_BROKEN_MESSAGE))
+			{
+				removeGameTimer(CANNON);
```

Comment:
Is there any case where this is needed? This message should only be received after the original 25 minutes has already elapsed, at which point `CANNON` would have run its full time and be culled anyway.

Reply:
I think I just wanted to make sure that the timer actually gets removed, given the states shouldn't overlap, and a lot of the timers drift slightly over time. It does look nicer imo to have only one of the two at a time.